src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorComparer.cs

```diff
@@ -47,11 +61,40 @@ public bool Equals(TagHelperDescriptor descriptorX, TagHelperDescriptor descript
         /// <returns>An <see cref="int"/> that uniquely identifies the given <paramref name="descriptor"/>.</returns>
         public int GetHashCode(TagHelperDescriptor descriptor)
         {
-            return HashCodeCombiner.Start()
-                                   .Add(descriptor.TagName, StringComparer.OrdinalIgnoreCase)
-                                   .Add(descriptor.TypeName, StringComparer.Ordinal)
-                                   .Add(descriptor.AssemblyName, StringComparer.Ordinal)
-                                   .CombinedHash;
+            return HashCodeCombiner
+                .Start()
+                .Add(descriptor.TagName, StringComparer.OrdinalIgnoreCase)
+                .Add(descriptor.TypeName, StringComparer.Ordinal)
+                .Add(descriptor.AssemblyName, StringComparer.Ordinal)
+                .Add(descriptor.RequiredAttributes)
```

Comment:
unfortunately this will be the hash code of the `IList<T>` instance.  two `Equals()` instances won't have the same hash code -- a big no-no.  probably need
``` c#
var combiner = HashCodeCombiner
  .Start()
  ...;
var orderedNames =
    descriptor.RequiredAttributes.OrderBy(attribute => attribute, StringComparer.OrdinalIgnoreCase);
foreach (var attributeName in orderedNames)
{
  combiner.Add(attributeName);
}

return combiner.CombinedHash;
```